tests/Integration/Behaviour/Features/Context/ProductFeatureContext.php

```diff
@@ -634,4 +634,16 @@ public function productWithNameProductisVirtual($productName)
         $this->products[$productName]->is_virtual = 1;
         $this->products[$productName]->save();
     }
+
+    /**
+     * @Given there is product with reference :productReference
+     */
+    public function thereIsProductOfTypeWithReference($productReference)
+    {
+        $productId = Product::getIdByReference($productReference);
+
+        if (!$productId) {
+            throw new \RuntimeException(sprintf('Product with reference "%s" does not exist.', $productReference));
```

Comment:
ProductNotFoundException?

Reply:
Hmm, it's a test, would it be handled differently? :thinking:

Reply:
It can use Core Exception, no?

Reply:
Same question, would it be handled differently? I mean, is there a reason not use native exceptions in behat tests? :thinking:

Reply:
Nope, as I know it will be handle exactly in the same way.
It's just to be consistent with the Core, and if we need to catch a specific exception during a Context.

Reply:
Hmmmm, @matks wdyt? I saw native exceptions were used everywhere in behat contexts.

Reply:
I have always used only RuntimeException in behat tests, because Exception seemed too generic 😅 basically we are writing assertions here, and we dont want to catch them. We want it to fail because it's intended to stop behat from working IMO